### ISSUE TITLE
Filtering collections by 24 letter string different then in mongo shell

### ISSUE THREAD

**Issue by msuchecki** [SoftInstigate/restheart#10]
The thing is that when filter collection with value that looks like ObjectId you replace it with new ObjectId (CollectionDAO.replaceObjectIds).

Example - I have a document in communites collection:
```
{
    "_id" : ObjectId("54a1535a60b2dd6e5353226b"),
    "name" : "Name 1",
    "description" : "Description 1",
    "_etag" : ObjectId("54b50a9d60b2174f23c91e4a"),
    "_created_on" : "2014-12-29T13:12:58Z",
    "nip" : "234-234-12-13",
    "administrator" : {
        "_id" : "548edf2560b2bb29fe2d6796",
        "name" : "Name 2",
        "avatar" : "../images/avatar-01.svg"
    }
}
```
I can filter collection using Mongo shell like so:
db.communities.find({"administrator._id": "548edf2560b2bb29fe2d6796"})

But I cant do the same with REST.

Not sure what to do with it. You should either replace ObjectId like values when saving document or not replacing it at all.

**Comment by ujibang** [comment]
you are right, it is a bug. 

I'm going to replace any ObjectId like values as a ObjectId when saving.

**Comment by msuchecki** [comment]
As far as I see id you use ObjectId in an attribute you get following output: 

```
"administrator":{  
      "_id":{  
         "$oid":"548edf3160b2bb29fe2d6798"
      },
      "name":"Zdzisława Bomba",
      "avatar":"../images/avatar-02.svg"
   }
```

How to do the filtering in this case? I tried:
```
db.communities.find({"administrator._id.$oid": "548edf3160b2bb29fe2d6798"})?
```
without any luck.

**Comment by ujibang** [comment]
fixed in current development branch. will be included in upcoming 0.9.8 release.

- On POST and PUT, any string that is a valid ObjectId will be written in mongodb as an ObjectId. 
- viceversa, On GET, the resource representation includes ObjectId properties as a strings (not {"$oid: "..."})
- any value in a filter expression that is a valid ObjectId will be taken into account as an ObjectId. 

I POSTed your example document and the following request now works.

    GET http://127.0.0.1:8080/db/coll?filter={'administrator._id':'548edf2560b2bb29fe2d6796'}

**Comment by ujibang** [comment]
hi, I have some news about this.

in the current develop branch (so available from upcoming 0.9.8 that is just some testing away) we added two query parameters:

* detect_oids (true|false with default: true) - if true it works as before, so replacing any string that is a valid ObjectId with an instance of this class (both on filters and json payload). You can now with off this by setting the param to false
* doc_id_type (string_objectid, string, int, long, float, double) that allows to specify the type of the _id

so now GET /db/coll/10?doc_id_type=int will work if the id is a number!
